prow/config/jobs.go

```diff
@@ -111,6 +111,8 @@ type JobBase struct {
 	Annotations map[string]string `json:"annotations,omitempty"`
 	// ReporterConfig provides the option to configure reporting on job level
 	ReporterConfig *prowapi.ReporterConfig `json:"reporter_config,omitempty"`
+	// RerunPermissions specifies who can rerun the job
+	RerunPermissions *prowapi.RerunPermissions `json:"rerun_permissions,omitempty"`
```

Comment:
I don't think this needs to be `omitempty` -- the uninitialized struct works fine, right? `AllowAnyone` will be false and the two lists will be empty

Reply:
I think it's nicer when we serialize jobs to omit the extra text if it's not necessary